### PR TITLE
QCportal cookbook

### PR DESCRIPTION
**Pull request by mattwelborn**
This PR adds a cookbook of miscellaneous ways to get data from QCArchive via portal.

### CODE REVIEW

File: cookbook/cookbook.ipynb [hunk 1]
```diff
@@ -0,0 +1,2474 @@
+{
```
Comment (dgasmith) [inline cookbook/cookbook.ipynb]: Maybe call this `qcp_cookbook_v1` so that we can extend later?

File: index.rst [hunk 2]
```diff
@@ -26,3 +26,10 @@ or with ``pip``::
    basic_examples/getting_started.ipynb
    basic_examples/reaction_datasets.ipynb
    basic_examples/torsiondrive_datasets.ipynb
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Cookbook
+
+
+   cookbook/cookbook.ipynb
```
Comment (dgasmith) [inline index.rst]: I imagined setting up examples like [Dask Examples](https://examples.dask.org). Would this structure be better?

Comment (mattwelborn) [reply index.rst]: Sure. To clarify, would the current notebook be split into:

* Get molecules and geometries
* Dataset and ReactionDataset: list available methods, basis sets, programs, etc.
* Get all calculations done on a molecule

Comment (dgasmith) [reply index.rst]: Lets give it a go.

File: cookbook/cookbook.ipynb (hunk 1 above)
Comment (mattwelborn) [reply cookbook/cookbook.ipynb]: Does splitting the cookbook resolve this suggestion?

Comment (dgasmith) [reply cookbook/cookbook.ipynb]: Yes, lets merge.

**Pull request merged**